File: kitsune/test/driver/fp-contract.f90
! REQUIRES: kitfc
!
! ------------------------------------------------------------------------------
! Unlike the other Kitsune frontends, the default FP contract value is the same
! as flang, but that default is not the same as clang's.
!
! RUN: %flang -fc1 -emit-mlir -o - %s 2>&1 \
! RUN:     | FileCheck %s -check-prefix CONTRACT-FAST
!
! ------------------------------------------------------------------------------
! Check that kitfc uses the same defaults as flang.
!
! RUN: %kitfc -fc1 -emit-mlir -o - %s 2>&1 \
! RUN:     | FileCheck %s -check-prefix CONTRACT-FAST
!
! CONTRACT-FAST: fastmath<contract>
!
! ------------------------------------------------------------------------------
! -ffp-contract=on is not supported in flang and maps to -ffp-contract=off. But
! we do not allow it at all in Kitsune. -ffp-contract=fast-honor-pragmas is not
! supported by Kitsune in Fortran.
!
! RUN: %flang -### -ffp-contract=on %s 2>&1 \
! RUN:     | FileCheck %s -check-prefix ON_TO_OFF
!
! RUN: not %kitfc -### -ffp-contract=on %s 2>&1 \
! RUN:     | FileCheck %s -check-prefix ERROR
!
! RUN: not %kitfc -### -ffp-contract=fast-honor-pragmas %s 2>&1 \
! RUN:     | FileCheck %s --check-prefix ERROR
!
! ON_TO_OFF: warning: the argument 'on' is not supported for option 'ffp-contract='. Mapping to 'ffp-contract=off'
! ERROR: error: unsupported argument '{{.+}}' to option '-ffp-contract=' for frontend '{{.+}}'

end program

File: kitsune/test/driver/fp-contract.c
// The default FP contract value is different from clang's defaults. The new
// default is only used if a tapir target is specified.
//
// Check that the defaults have not changed when running without a Kitsune
// frontend
//
// RUN: %clang -### %s 2>&1 | FileCheck %s -check-prefix CONTRACT-ON
//
// When running with a Kitsune frontend, this value should be ON. Also check
// that the value can be overridden if required.
//
// RUN: %kitcc -### %s 2>&1 | FileCheck %s -check-prefix CONTRACT-ON
// RUN: %kitcc -### -ffp-contract=off %s 2>&1 \
// RUN:     | FileCheck %s -check-prefix CONTRACT-OFF
// RUN: %kitcc -### -ffp-contract=off -ftapir=serial %s 2>&1 \
// RUN:     | FileCheck %s --check-prefix CONTRACT-OFF
// RUN: %kitcc -### -ffp-contract=fast %s 2>&1 \
// RUN:     | FileCheck %s -check-prefix CONTRACT-FAST
// RUN: %kitcc -### -ffp-contract=fast -ftapir=serial %s 2>&1 \
// RUN:     | FileCheck %s --check-prefix CONTRACT-FAST
//
// CONTRACT-OFF: "-ffp-contract=off"
// CONTRACT-ON: "-ffp-contract=on"
// CONTRACT-FAST: "-ffp-contract=fast"
